File: 16_April/Iteration.py
def iteration_prac():
    num_input = input("Enter: ")
    num = int(num_input)
    print("x: ", num)
    for i in range(1, num):
        if num % i == 0:
            print(str(i) + " is a factor of " + str(num))
        else:
            print(str(i) + " is not a factor of " + str(num))


def iteration_prac_while():
    num_input = input("Enter: ")
    num = int(num_input)
    i = num
    print("x: ", num)
    while num > 0:
        if num % i == 0:
            print(str(i) + " is a factor of " + str(num))
        else:
            print(str(i) + " is not a factor of " + str(num))
        i -= 1

iteration_prac_while()
iteration_prac()

File: 16_April/Iteration.java
import java.util.*;
import java.lang.System;

class Iteration {
    public static void main(String[] args) {
        Scanner obj1 = new Scanner(System.in);
        System.out.print("Enter Number: ");
        Integer num = obj1.nextInt();
        System.out.println("x: " + num);

        for (int i = num; num > 0; num--)
            if (i % num == 0) {
                System.out.println(num + " is a factor of " + i);
            } else if (num == 0) {
                break;
            } else {
                System.out.println(num + " is not factor of " + i);
            }

        // int i = num;
        // while (num > 0) {
        //     if (i % num == 0) {
        //         System.out.println(num + " is a factor of " + i);
        //         num--;
        //     } else if (num == 0) {
        //         break;
        //     } else {
        //         System.out.println(num + " is not factor of " + i);
        //         num--;
        //     }
        // }
    }

}
